src/analysis/retail/priest/shadow/modules/talents/Archon/Concentrated Infusion.tsx

```diff
@@ -0,0 +1,40 @@
+import TALENTS from 'common/TALENTS/priest';
+import { Options } from 'parser/core/Module';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { ApplyBuffEvent, RemoveBuffEvent } from 'parser/core/Events';
+import Haste from 'parser/shared/modules/Haste';
+
+import { ARCHON_CONCENTRATED_INFUSION_HASTE } from '../../../constants';
+
+class ConcentratedInfusion extends Analyzer {
+  static dependencies = {
+    haste: Haste,
+  };
+
+  protected haste!: Haste;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS.CONCENTRATED_INFUSION_TALENT);
+
+    this.addEventListener(
```

Comment:
making a note that pre-TWW this would not have worked right because refreshes of PI didn't log a new PI buff. however, they have fixed that so this should be correct for all TWW logs (unless they un-fix it :upside_down_face:)

Reply:
I hadn't even considered a PI refresh being a problem, thank goodness that's fixed (for now).